### PR TITLE
Email service

### PR DESCRIPTION
**Pull request by alhansrisuk**
Creating new pull request to sync fork's branch.

### CODE REVIEW

**Comment by StoneyJackson** [comment]
Closes #54

**Comment by StoneyJackson** [comment]
Please remove the pyc files.

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
```
Comment (KevinTran000) [inline yarn.lock:4]: Is this lock file supposed to be tracked/committed in?

Comment (StoneyJackson) [reply yarn.lock:4]: I think it typically is. https://yarnpkg.com/lang/en/docs/yarn-lock/#toc-check-into-source-control

**Comment by StoneyJackson** [comment]
Your team told me that this branch works on some of your systems and not on others. You showed me the error had to do with code in a 3rd party library and had nothing to do with the data you passed to the library.

We explored the version number of the 3rd party library (flask-sendgrid I think), and they were the same. However, we did not compare the packages that fask-sendgrid depends on. Let's do that.

Please post the results of `pip freeze`. @alhansrisuk I think you had the working installation. So can you please post your `pip freeze` result. And I need one or more of you whose installation is giving errors to post their `pip freeze` result. Also, please post the complete error message you are getting.

If we find that there is a difference in dependency versions, we might have found our culprit. If not, then we will know I'm wrong and we should look elsewhere.

It's important to get this documented here so that if we ever encounter this again, others will have a record of what we did.

Last note, a description of the problem I *think* we are having is described in [this article](https://www.kennethreitz.org/essays/a-better-pip-workflow) along with a solution. But let's confirm you really are having this problem. FYI, I'm suggesting that we adopt this practice anyway in #86 .

**Comment by alhansrisuk** [comment]
alberthansrisuk:bear-necessities-market $ pip freeze
alembic==1.0.7
aniso8601==1.3.0
astroid==2.2.5
atomicwrites==1.3.0
attrs==18.2.0
bcrypt==3.1.6
blinker==1.4
cffi==1.12.2
click==6.7
Flask==1.0.2
Flask-API==1.1
Flask-Bcrypt==0.7.1
Flask-Cors==3.0.7
Flask-JWT==0.3.2
Flask-JWT-Extended==3.17.0
Flask-Mail==0.9.1
flask-marshmallow==0.9.0
Flask-Migrate==2.3.1
Flask-RESTful==0.3.6
flask-restplus==0.12.1
Flask-Script==2.0.6
Flask-SendGrid==0.6
Flask-SQLAlchemy==2.3.2
isort==4.3.16
itsdangerous==0.24
Jinja2==2.10
jsonschema==3.0.1
lazy-object-proxy==1.3.1
Mako==1.0.7
MarkupSafe==1.0
marshmallow==2.18.1
mccabe==0.6.1
more-itertools==6.0.0
pluggy==0.8.1
py==1.7.0
pycparser==2.19
PyJWT==1.4.2
pylint==2.3.1
pyrsistent==0.14.11
pytest==4.2.1
python-dateutil==2.6.1
python-editor==1.0.4
python-http-client==3.1.0
pytz==2017.3
sendgrid==5.6.0
six==1.11.0
SQLAlchemy==1.1.15
typed-ast==1.3.1
ujson==1.35
virtualenv==16.4.0
Werkzeug==0.14.1
wrapt==1.11.1

Here is my result for `pip freeze`.

**Comment by StoneyJackson** [comment]
Awesome. Thanks!

@ARule Can we get your `pip freeze` result and the error message you've been getting?

**Comment by StoneyJackson** [comment]
What if you all try what I outlined in #86 and if it works, then we can close it and probably #97 too.

**Comment by dgsienko** [comment]
**This is the error @ARule and I are getting:** 

<class 'backend.config.ProductionConfig'>
Traceback (most recent call last):
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/bin/flask", line 12, in <module>
    sys.exit(main())
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask/cli.py", line 894, in main
    cli.main(args=args, prog_name=name)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask/cli.py", line 557, in main
    return super(FlaskGroup, self).main(*args, **kwargs)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask/cli.py", line 767, in run_command
    app = DispatchingApp(info.load_app, use_eager_loading=eager_loading)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask/cli.py", line 293, in __init__
    self._load_unlocked()
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask/cli.py", line 317, in _load_unlocked
    self._app = rv = self.loader()
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask/cli.py", line 377, in load_app
    raise_if_not_found=False)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask/cli.py", line 235, in locate_app
    __import__(module_name)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/wsgi.py", line 12, in <module>
    application = create_app(ProductionConfig)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/backend/app.py", line 42, in create_app
    create_mail_server(app)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/backend/app.py", line 81, in create_mail_server
    app.mail = SendGrid(app)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask_sendgrid.py", line 24, in __init__
    self.init_app(app)
  File "/Users/dillonsienko/Desktop/Local BNM Work/BEAR-Necessities-Market/env/lib/python3.6/site-packages/flask_sendgrid.py", line 48, in init_app
    self.client = SendGridAPIClient(apikey=self.api_key).client
TypeError: __init__() got an unexpected keyword argument 'apikey'

**And these are the results from my pip freeze:** 

alembic==1.0.7
aniso8601==1.3.0
atomicwrites==1.3.0
attrs==19.1.0
bcrypt==3.1.6
blinker==1.4
cffi==1.12.2
click==6.7
Flask==1.0.2
Flask-API==1.1
Flask-Bcrypt==0.7.1
Flask-Cors==3.0.7
Flask-JWT==0.3.2
Flask-JWT-Extended==3.18.1
Flask-Mail==0.9.1
flask-marshmallow==0.9.0
Flask-Migrate==2.3.1
Flask-RESTful==0.3.6
flask-restplus==0.12.1
Flask-Script==2.0.6
Flask-SendGrid==0.6
Flask-SQLAlchemy==2.3.2
itsdangerous==0.24
Jinja2==2.10
jsonschema==3.0.1
Mako==1.0.7
MarkupSafe==1.0
marshmallow==2.19.2
more-itertools==7.0.0
pluggy==0.9.0
py==1.8.0
pycparser==2.19
PyJWT==1.4.2
pyrsistent==0.14.11
pytest==4.4.1
python-dateutil==2.6.1
python-editor==1.0.4
python-http-client==3.1.0
pytz==2017.3
sendgrid==6.0.4
six==1.11.0
SQLAlchemy==1.1.15
ujson==1.35
Werkzeug==0.14.1

**Comment by StoneyJackson** [comment]
OK, there are differences between your dependencies. I think this is a larger problem that needs to be solved for the project. I think we need to fix #86 first, and then return to this. So any help you can help there would be great.

**Comment by StoneyJackson** [comment]
OK, requirements.txt and requirements-to-freeze.txt are now in master. I've updated our shop fork. Sync your branch and you be good to go.

**Review by StoneyJackson — DISMISSED** [review]

**Review by hawzie197 — APPROVED** [review]
Good to go after commenting out mail.py send_mail()

**Review by StoneyJackson — APPROVED** [review]

**Pull request merged**